smartmeter/display.cpp

```diff
@@ -123,6 +127,23 @@ void Display::displayLogo()
     display->display();
 }
 
+void Display::run()
+{
+    static uint counter = 0;
+    static uint screenId = 0;
+    static uint lastScreenId = -1;
+    static const uint numOfScreens = screens.size();
+
+    counter++;
+    screenId = (int)counter / (targetFps * delayBetweenFrames);
+    screenId = screenId % numOfScreens;
+
+    if (screenId != lastScreenId) {
+        screens[screenId]();
+        lastScreenId = screenId;
+    }
+}
```

Comment:
Did you already test this block to see if the logic is working?

Everything else seens fine to me.

Reply:
Yep, here you can get the math.
K = n*16/(16*5)
id = K%3
id = (n/5)%3

Reply:
Ok